uniffi_bindgen/src/bindings/swift/templates/ObjectTemplate.swift

```diff
@@ -9,30 +9,43 @@ public protocol {{ obj.name() }}Protocol {
     {% endfor %}
 }
 
-public class {{ obj.name()|class_name_swift }}: {{ obj.name() }}Protocol {
-    private let handle: UInt64
+public class {{ obj.name()|class_name_swift }}: {{ obj.name() }}Protocol, Equatable, Hashable {
+    private let pointer: UnsafeMutableRawPointer
 
-    private init(fromRawHandle handle: UInt64) {
-        self.handle = handle
+    // TODO: We'd like this to be `private` but for Swifty reasons,
+    // we can't implement `ViaFfi` without making this `required` and we can't
+    // make it `required` without making it `public`.
+    required init(unsafeFromRawPointer pointer: UnsafeMutableRawPointer) {
+        self.pointer = pointer
     }
 
     {%- match obj.primary_constructor() %}
     {%- when Some with (cons) %}
     public convenience init({% call swift::arg_list_decl(cons) -%}) {% call swift::throws(cons) %} {
-        self.init(fromRawHandle: {% call swift::to_ffi_call(cons) %})
+        self.init(unsafeFromRawPointer: {% call swift::to_ffi_call(cons) %})
     }
     {%- when None %}
     {%- endmatch %}
 
     deinit {
         try! rustCall(UniffiInternalError.unknown("deinit")) { err in
-            {{ obj.ffi_object_free().name() }}(handle, err)
+            {{ obj.ffi_object_free().name() }}(pointer, err)
         }
     }
 
+    // Object equality is pointer identity.
+    // TODO: perhaps we should delegate to Rust for this sort of thing?
+    public static func ==(lhs: {{ obj.name()|class_name_swift }}, rhs: {{ obj.name()|class_name_swift }}) -> Bool {
+        return lhs.pointer == rhs.pointer
+    }
+
+    public func hash(into hasher: inout Hasher) {
+        hasher.combine(self.pointer)
+    }
```

Comment:
Here I've implemented `Equatable` and `Hashable` for Object classes, mostly so that they can appear as fields in Records, which are in turn `Equatable` and `Hashable`. An alternative would be to only implement `Equatable` and `Hashable` for Records that do not contain any Object instances.

Reply:
> Recommend we don't implement Equatable and Hashable until we have a better sense of demand for this.

Aye, thanks. I think I knew that this was most likely the right thing to do but just didn't want to deal with it in this PR. I'll update.

Reply:
Specifically, what I've done is:

* Do not implement `Equatable/Hashable` for `Object` types.
* Update `Record` and `Enum` types so that they only implement `Equatable/Hashable` if they do not contain any `Object` type fields.